### PR TITLE
Fix for issue 5925: constant arrays should be nonlinear.

### PR DESCRIPTION
**Pull request by barrettcw**
The "nonlinear" optimization did not take constant arrays into account.  This fixes the issue.

### CODE REVIEW

**Review by 4tXJ7f — COMMENTED** [review]
This is a great find! Can you add the benchmark from the issue as a regression?

**Comment by 4tXJ7f** [comment]
I can also add it if you don't have the cycles.

File: test/regress/cli/regress0/arrays/issue5925-2.smt2 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+; COMMAND-LINE: --no-check-proofs
```
Comment (4tXJ7f) [inline test/regress/cli/regress0/arrays/issue5925-2.smt2:1]:
```suggestion
; DISABLE-TESTER: proof
```

File: test/regress/cli/regress0/arrays/issue5925.smt2 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+; COMMAND-LINE: --no-check-proofs
```
Comment (4tXJ7f) [inline test/regress/cli/regress0/arrays/issue5925.smt2:1]:
```suggestion
; DISABLE-TESTER: proof
```

**Review by 4tXJ7f — APPROVED** [review]
LGTM, I just added suggestions for using the new commands for disabling testers.

**Comment by ajreynol** [comment]
Proofs were failing due to the use of arrays of Booleans, which need special care due to how the equality engine handles explanations for Boolean terms. I will fix this in a separate PR.

**Pull request merged**